crypto3/libs/zk/include/nil/crypto3/zk/snark/systems/plonk/placeholder/verifier.hpp

```diff
@@ -330,6 +330,31 @@ namespace nil {
                                 ++j;
                             }
                         }
+                        typename FieldType::value_type mask_value = FieldType::value_type::one() -
+                            proof.eval_proof.eval_proof.z.get(FIXED_VALUES_BATCH, common_data.permuted_columns.size() * 2, 0) -
+                            proof.eval_proof.eval_proof.z.get(FIXED_VALUES_BATCH, common_data.permuted_columns.size() * 2 + 1, 0);
+                        typename FieldType::value_type shifted_mask_value = FieldType::value_type::one() -
+                            proof.eval_proof.eval_proof.z.get(FIXED_VALUES_BATCH, common_data.permuted_columns.size() * 2, 1) -
+                            proof.eval_proof.eval_proof.z.get(FIXED_VALUES_BATCH, common_data.permuted_columns.size() * 2 + 1, 1);
+
+                        // All rows selector
+                        {
+                            auto key = std::make_tuple( -1, 0, plonk_variable<typename FieldType::value_type>::column_type::selector);
```

Comment:
Can you create a const variable for value -1, just like you did in gate argument with PLONK_SPECIAL_SELECTOR_ALL_USABLE_ROWS_SELECTED. Maybe we need to do something like

static const int PLONK_SPECIAL_SELECTOR_ALL_USABLE_ROWS_SELECTOR_INDEX = -1;

Reply:
Done. Now constants are used in this place